android/anc/src/main/java/org/smartregister/fhircore/anc/data/madx/NonAncPatientRepository.kt

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2021 Ona Systems, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.smartregister.fhircore.anc.data.madx
+
+import com.google.android.fhir.FhirEngine
+import com.google.android.fhir.logicalId
+import com.google.android.fhir.search.search
+import kotlinx.coroutines.withContext
+import org.hl7.fhir.r4.model.CarePlan
+import org.hl7.fhir.r4.model.Condition
+import org.hl7.fhir.r4.model.Encounter
+import org.hl7.fhir.r4.model.Patient
+import org.smartregister.fhircore.anc.AncApplication
+import org.smartregister.fhircore.anc.data.madx.model.AllergiesItem
+import org.smartregister.fhircore.anc.data.madx.model.AncPatientDetailItem
+import org.smartregister.fhircore.anc.data.madx.model.AncPatientItem
+import org.smartregister.fhircore.anc.data.madx.model.CarePlanItem
+import org.smartregister.fhircore.anc.data.madx.model.ConditionItem
+import org.smartregister.fhircore.anc.data.madx.model.EncounterItem
+import org.smartregister.fhircore.anc.data.madx.model.UpcomingServiceItem
+import org.smartregister.fhircore.engine.util.DateUtils.makeItReadable
+import org.smartregister.fhircore.engine.util.DefaultDispatcherProvider
+import org.smartregister.fhircore.engine.util.DispatcherProvider
+import org.smartregister.fhircore.engine.util.extension.due
+import org.smartregister.fhircore.engine.util.extension.extractAge
+import org.smartregister.fhircore.engine.util.extension.extractGender
+import org.smartregister.fhircore.engine.util.extension.extractName
+import org.smartregister.fhircore.engine.util.extension.overdue
+
+class NonAncPatientRepository(
```

Comment:
I don't think we want a specific non-ANC repository. What if we move the [`PatientRepository`](https://github.com/opensrp/fhircore/blob/main/android/eir/src/main/java/org/smartregister/fhircore/eir/data/PatientRepository.kt) from EIR to engine, extend that, and allow a way to filter subsets on query to specify non-ANC dynamically?

@ellykits looks like you wrote the PatientRepository, can you weigh in here on a way to make this more generic?

also CC @ekigamba @maimoonak @ndegwamartin for input